### PR TITLE
2.11.0

### LINKED ISSUE #891: Native types

### LINKED ISSUE #924: Datasource provider array update

### LINKED ISSUE #948: Capitalization consistency for product maturity stages

### PR DESCRIPTION
**Pull request by mhwelander**

### CODE REVIEW

File: content/200-concepts/100-components/07-preview-features/200-native-types/index.mdx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+title: 'Extended native types support'
+metaTitle: 'Extended native types support (Reference)'
+metaDescription: 'Overview of the native types preview feature.'
+---
+
+<TopBlock>
+
+Extended native database type support is a preview feature in [2.11.0](https://github.com/prisma/prisma/releases/2.11.0). This preview includes the following features:
+
+* Introspection adds **type attributes** to each field that **enrich your schema** with information about the underlying native database type
```
Comment (matthewmueller) [inline content/200-concepts/100-components/07-preview-features/200-native-types/index.mdx:11]: "enrich" is such a good word here 💯

**Review by matthewmueller — APPROVED** [review]
Looks great to me 🙌

File: content/200-concepts/100-components/07-preview-features/200-native-types/index.mdx (hunk 1 above)
Comment (mhwelander) [reply content/200-concepts/100-components/07-preview-features/200-native-types/index.mdx:11]: I stole it from @nikolasburk 🙏

**Comment by mhwelander** [comment]
<s>Note to self: Update list of supported types w/ blob family</s> ✔

File: content/200-concepts/100-components/02-prisma-client/180-middleware/index.mdx [hunk 2]
```diff
@@ -51,12 +51,19 @@ Use cases for middleware include:
...
+  action: PrismaAction
   args: any
   dataPath: string[]
   runInTransaction: boolean
 }
+
+export declare const ModelName: {
+  User: 'User',
+  Post: 'Post'
+};
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/180-middleware/index.mdx]: I guess without `;` here as well?
```suggestion
}
```

File: content/200-concepts/100-components/07-preview-features/200-native-types/index.mdx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+Response "Some of the requested preview features are not yet allowed in migration engine. Please remove them from your data model before using migrations. (blocked: `nativeTypes`)"
+```
+
+### <inlinecode>db push</inlinecode> not yet supported
+
+`db push` uses the migration engine, which does not support native database type attributes yet.
+
+### SQL Server not yet supported
+
+You cannot use native database type attributes with the [SQL Server preview](../sql-server).
```
Comment (janpio) [inline content/200-concepts/100-components/07-preview-features/200-native-types/index.mdx]:
```suggestion
You cannot use native database type attributes with the [SQL Server preview](../sql-server) yet.
```

File: content/200-concepts/100-components/07-preview-features/200-native-types/index.mdx [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+Prisma Migrate does not yet support native database type attributes. If you try to `prisma migrate` or `prisma db push`, you will see the following error:
+
+```terminal
+Response "Some of the requested preview features are not yet allowed in migration engine. Please remove them from your data model before using migrations. (blocked: `nativeTypes`)"
+```
+
+### <inlinecode>db push</inlinecode> not yet supported
+
+`db push` uses the migration engine, which does not support native database type attributes yet.
```
Comment (janpio) [inline content/200-concepts/100-components/07-preview-features/200-native-types/index.mdx]:
```suggestion
`db push` uses the Migration Engine, which does not support native database type attributes yet.
```

File: content/200-concepts/100-components/07-preview-features/200-native-types/native-types-mappings.mdx [hunk 5]
```diff
@@ -0,0 +1,378 @@
+---
+title: 'Type mapping reference'
+metaTitle: 'Type mapping reference (Reference)'
+metaDescription: 'A list of Prisma features currently in preview.'
```
Comment (janpio) [inline content/200-concepts/100-components/07-preview-features/200-native-types/native-types-mappings.mdx]: outdated description

File: content/200-concepts/100-components/07-preview-features/200-native-types/native-types-mappings.mdx [hunk 6]
```diff
@@ -0,0 +1,378 @@
...
+
+| Native database type | Native database type attribute (Preview) | Introspection | 
+| :------------------------ | :--------------------------------- | :------------------| 
+| `text` (default type)     | `@db.Text`                                | ✔️                
+| `char`                    | `@db.Char(x)`                         | ✔️ ||
+| `varchar`                 | `@db.VarChar(n)`                      | ✔️||
+| `bit`                     | `@db.Bit`                          | ✔️||
+| `varbit`                  | `@db.VarBit`                       | ✔️||
+| `uuid`                    | `@db.Uuid`                         | ✔️||
+| `xml`                    | `@db.Xml`                         | ✔️||
```
Comment (janpio) [inline content/200-concepts/100-components/07-preview-features/200-native-types/native-types-mappings.mdx]:
```suggestion
| :------------------------ | :--------------------------------- | :------------------| 
| `text` (default type)     | `@db.Text`                         | ✔️ |               
| `char`                    | `@db.Char(x)`                      | ✔️ |
| `varchar`                 | `@db.VarChar(n)`                   | ✔️ |
| `bit`                     | `@db.Bit`                          | ✔️ |
| `varbit`                  | `@db.VarBit`                       | ✔️ |
| `uuid`                    | `@db.Uuid`                         | ✔️ |
| `xml`                     | `@db.Xml`                          | ✔️ |
```

File: content/200-concepts/100-components/07-preview-features/200-native-types/native-types-mappings.mdx [hunk 7]
```diff
@@ -0,0 +1,378 @@
...
+The following table lists the MySQL native database types that map to Prisma's `String` scalar type:
+
+| Native database type | Native database type attribute | Introspection                       | 
+| :------------------------ | :--------------------------------- | :--------------------------------- | 
+| `VARCHAR` (default type)  | `@db.VarChar(191)`                      | ✔️ | 
+| `TEXT`                    | `@db.Text`                         | ✔️ |                                  |
+| `CHAR`                    | `@db.Char`                         |✔️||
+| `TINYTEXT`                | `@db.TinyText`                     |✔️||
+| `MEDIUMTEXT`              | `@db.MediumText`                   |✔️||
+| `LONGTEXT`                | `@db.LongText`                     |✔️||
```
Comment (janpio) [inline content/200-concepts/100-components/07-preview-features/200-native-types/native-types-mappings.mdx]:
```suggestion
| Native database type | Native database type attribute | Introspection                    | 
| :------------------------ | :--------------------------------- | :---------------------------- |
| `VARCHAR` (default type)  | `@db.VarChar(191)`                 | ✔️ | 
| `TEXT`                    | `@db.Text`                         | ✔️ |                                  
| `CHAR`                    | `@db.Char`                         | ✔️ |
| `TINYTEXT`                | `@db.TinyText`                     | ✔️ |
| `MEDIUMTEXT`              | `@db.MediumText`                   | ✔️ |
| `LONGTEXT`                | `@db.LongText`                     | ✔️ |
```

File: content/200-concepts/100-components/07-preview-features/200-native-types/index.mdx (hunk 3 above)
Comment (mhwelander) [inline content/200-concepts/100-components/07-preview-features/200-native-types/index.mdx]:
```suggestion
You cannot use native database type attributes with the [SQL Server preview](../sql-server) yet.
```

**Pull request merged**